### ISSUE TITLE
Naming of env variables seems to be inconsistent

### ISSUE THREAD

**Issue by Zelldon** [zeebe-io/zeebe#4185]
**Description**

For me it is a bit unclear why sometimes we have underscores and sometimes not, e.g.:

```
      # This setting can also be overridden using the environment variable ZEEBE_BROKER_NETWORK_MAXMESSAGESIZE.
      # maxMessageSize: 4MB
```

**Comment by pihme** [comment]
tldr: Underscores appear whenever you have a nesting in your object graph. 

Except for the top level `zeebe_broker` - there it is just a matter of aesthetics. It used to be `ZEEBEBROKER` but several people didn't like this.

Long story:
Spring gives you a certain amount of leniency; https://docs.spring.io/spring-boot/docs/current/reference/html/spring-boot-features.html#boot-features-external-config-relaxed-binding

So you might be able to use `ZEEBE_BROKER_NETWORK_MAX_MESSAGE_SIZE` and it might get mapped properly. The word "might" is the culprit here. And the devil is in the detail.

After playing around with the different ways to write the environment variables the structure now documented in the code proofed to be the most foolproof one, and the one most natural to the look of environment variables.

**Comment by Zelldon** [comment]
ok thanks @pihme for clarification :+1: 

might be also the question if the `broker` in the name is really necessary or not

**Comment by pihme** [comment]
the broker is there to disambiguate it from `zeebe-gateway`. Otherwise the two namespaces would clash and it would be more confusing.

**Comment by Zelldon** [comment]
But why is this necessary?

I mean if you specify the gateway in the broker then you want a embedded gateway. Which means it is **inside** the broker right? So instead of having

```yaml
zeebe:
  gateway:
  broker:
   network:
   data:
   cluster:
```

I would expect:

```yaml
zeebe:
  broker:
    gateway:
    network:
    data:
    cluster:
```

Then we could just skip the broker level and move everything in the zeebe section and have just the subsection for gateway

```yaml
zeebe:
  gateway:
  network:
  data:
  cluster:
```

**Comment by pihme** [comment]
When I talked about gateway, I did mean the standalone gateway. The namespace over standalone broker and standalone gateway is as follows:

```
zeebe:
  broker:  //standalone broker
    gateway: //embedded gateway
      network:
      cluster:
      threads:
      ...
    network:
    data:
    cluster:
  gateway; //standalone gateway
   network: 
   cluster:
   threads:
```
It was a constraint to not change the structure of the existing configuration as it is. 

Given that, the embedded gateway is the odd case. The configuration options for the embedded gateway and the standalone gateway are not exactly the same. Some configuration options for the embedded gateway are not advertised, but could theoretically be set. Other configuration options can be set, but will be overwritten anyway by other configuration settings of the broker configuration.

```
  public void init(final BrokerCfg globalConfig, final String brokerBase) {

    // configure gateway based on broker network settings
    final NetworkCfg networkCfg = globalConfig.getNetwork();

    init(networkCfg.getHost());

    // ensure embedded gateway can access local broker
    getCluster()
        .setContactPoint(SocketUtil.toHostAndPortString(networkCfg.getInternalApi().getAddress()));

    // configure embedded gateway based on broker config
    getNetwork().setPort(getNetwork().getPort() + (networkCfg.getPortOffset() * 10));
  }
```

I think your comments are valid.

However, I feel they come a little late for this release. 

If you want to suggest changes for the future, I feel this should be discussed more broadly within the team.
